File: link-mobile-app/iosApp/iosApp/UI/Components/LinkSwitch.swift
import SwiftUI

/// A styled toggle switch that matches the Link design system.
struct LinkSwitch: View {
    @Binding var isOn: Bool
    var isEnabled: Bool = true

    var body: some View {
        Toggle("", isOn: $isOn)
            .labelsHidden()
            .disabled(!isEnabled)
    }
}

File: link-mobile-app/composeApp/src/androidMain/kotlin/com/stripe/link/core/ui/component/LinkSwitch.kt
package com.stripe.link.core.ui.component

import androidx.compose.material.Switch
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier

/**
 * A styled toggle switch that matches the Link design system.
 *
 * @param checked Whether the switch is in the on state.
 * @param onCheckedChange Callback when the switch is toggled. Pass `null` to disable interaction.
 * @param modifier [Modifier] applied to the switch.
 * @param enabled Whether the switch is interactive.
 */
@Composable
fun LinkSwitch(
    checked: Boolean,
    onCheckedChange: ((Boolean) -> Unit)?,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
) {
    Switch(
        checked = checked,
        onCheckedChange = onCheckedChange,
        modifier = modifier,
        enabled = enabled && onCheckedChange != null,
    )
}
